sound/soc/sof/pm.c

```diff
@@ -335,8 +336,20 @@ int snd_sof_prepare(struct device *dev)
 		return 0;
 
 #if defined(CONFIG_ACPI)
-	if (acpi_target_system_state() == ACPI_STATE_S0)
+	system_target_state = acpi_target_system_state();
+	switch (system_target_state) {
```

Comment:
Use `acpi_target_system_state()` directly in the switch and then there is no need for the variable as it is otherwise unused?

Reply:
done, thanks for the suggestion